### DISCUSSION TITLE
How do I solve this?

### DISCUSSION THREAD

**Discussion by Nethrenial** [prisma/prisma#12905] [Q&A]
I've written a simple text search query with the Prisma client to search Usernames or emails in my users table. While it correctly works with usernames, it errors out saying, `Unexpected '@'`.I believe that's due to @ is a special symbol.Is there anyway around this, because I need to be able to search full email addresses. (Emails are correctly saved in the database). And query return results if I omit the part from @.Ex: in abc@gmail.com, query returns match if I just search abc.But If I use abc@gmail.com, it fails with this error.
```bash
 Error occurred during query execution:
 ConnectorError(
  ConnectorError { 
   user_facing_error: None, 
   kind: QueryError(Server(ServerError { 
                           code: 1064, message: "syntax error, 
                           unexpected '@',expecting $end", state: "42000" }
)) })
```
How do I solve this, I searched the documentation, but I couldn't find anything related to this.
My User model looks like this,
```prisma
model User {
  id              Int     @id @default(autoincrement())
  username        String  @db.VarChar(255)
  email           String  @db.VarChar(255)
  avatar          String? @db.VarChar(255)
  password        String  @default("1") @db.VarChar(255)
  friends         User[]  @relation("friends")
  friendsRelation User[]  @relation("friends")
  hashRT          String? @db.VarChar(255)

  @@fulltext([username])
  @@fulltext([email])
  @@fulltext([username, email])
  @@map("users")
}
```

**Comment by austincrim** [comment]
Hey @Nethrenial 👋 ,

What DB provider and Prisma version are you using? I was able to query successfully with an `@` sign using Postgres and Prisma 3.12.

**Reply by Nethrenial** [reply]
"prisma": "^3.12.0"
and provider is MYSQL

**Reply by austincrim** [reply]
Try wrapping your search term in double quotes:
```ts
let result = await prisma.user.findMany({
  where: {
    email: {
      search: '"foo@bar"'  // outer single quotes, search term wrapped in double quotes
    }
  }
})
```

**Reply by vladi-stevanovic** [reply]
👋 Hello @Nethrenial  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Reply by cdcarson** [reply]
@vladi-stevanovic I can confirm this is still an issue (or at least a documentation gap.) I'm using MySQL on PlanetScale.  

The issue is that Prisma seems to be using MySql's [Boolean full text search](https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html), not its [natural language search](https://dev.mysql.com/doc/refman/8.0/en/fulltext-natural-language.html).  I'm not familiar enough with MySQL search to have an opinion which is better, but I think the default assumption for most users would be that Prisma would use natural language -- it seems like it's the most salient, easiest use case.

For example, I can do this no problem...

```ts
const users = await db.$queryRaw`
  SELECT 
    userId, 
    email 
  FROM UserAccount 
  WHERE MATCH (email) AGAINST ('@example' IN NATURAL LANGUAGE MODE)`;
```
...without wrapping `@example.com`. Whereas this...

```ts
const users = await db.$queryRaw`
    SELECT
      userId,
      email
    FROM UserAccount
    WHERE MATCH (email) AGAINST ('@example' IN BOOLEAN MODE)`;
```
...results in an error because `@` is the "distance" operator in MySQL boolean search, e.g.... 

```sql
MATCH(col1) AGAINST('"word1 word2 word3" @8' IN BOOLEAN MODE)
```

Using @austincrim's solution of wrapping the query in double quotes seems to work OK, but I'd be worried about the situation where the query already contains a double quote. Prisma should definitely take care of this for the user.

Or, just switch to `IN NATURAL LANGUAGE MODE`.

**Reply by pantharshit00** [reply]
Hello @cdcarson 

I have created an issue in order to track this. https://github.com/prisma/prisma/issues/16537

Please follow that now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
